src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1269,13 +1270,15 @@ MergeTreeData::LoadPartResult MergeTreeData::loadDataPart(
         throw;
     }
 #if USE_AZURE_BLOB_STORAGE
-    catch (const Azure::Core::Http::TransportException &)
+    catch (const Azure::Core::RequestFailedException & e)
     {
+        LOG_DEBUG(log, "Failed to load data part {}, Exception {}", part_name, e.Message);
```

Comment:
We can keep the logs(atleast for now) as they will help to get the issue if part is detached by some other exception